generator/Makefile

```diff
@@ -0,0 +1,115 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+MIBDIR := mibs
+
+APC_URL        := 'https://download.schneider-electric.com/files?p_File_Name=powernet426.mib'
+ARISTA_URL     := https://www.arista.com/assets/data/docs/MIBS
+CISCO_URL      := 'ftp://ftp.cisco.com/pub/mibs/v2/v2.tar.gz'
+MIKROTIK_URL   := 'http://download2.mikrotik.com/Mikrotik.mib'
+PALOALTO_URL   := 'https://www.paloaltonetworks.com/content/dam/pan/en_US/assets/zip/technical-documentation/snmp-mib-modules/PAN-MIB-MODULES-7.0.zip'
+SERVERTECH_URL := 'ftp://ftp.servertech.com/Pub/SNMP/sentry3/Sentry3.mib'
+SYNOLOGY_URL   := 'https://global.download.synology.com/download/Document/MIBGuide/Synology_MIB_File.zip'
+UBNT_DL_URL    := http://dl.ubnt-ut.com/snmp
+UBNT_AIROS_URL := https://dl.ubnt.com/firmwares/airos-ubnt-mib/ubnt-mib.zip
+
+
+.DEFAULT: all
+
+.PHONY: all
+all: mibs
+
+clean:
+	rm -v \
+		$(MIBDIR)/* \
+		$(MIBDIR)/.cisco_v2 \
+		$(MIBDIR)/.paloalto_panos \
+		$(MIBDIR)/.synology
+
+.PHONY: mibs
+mibs: mib-dir \
+  $(MIBDIR)/apc-powernet-mib \
+  $(MIBDIR)/ARISTA-ENTITY-SENSOR-MIB \
+  $(MIBDIR)/ARISTA-SMI-MIB \
+  $(MIBDIR)/ARISTA-SW-IP-FORWARDING-MIB \
+  $(MIBDIR)/.cisco_v2 \
+  $(MIBDIR)/.paloalto_panos \
+  $(MIBDIR)/servertech-sentry3-mib \
+  $(MIBDIR)/.synology \
+  $(MIBDIR)/UBNT-MIB \
+  $(MIBDIR)/UBNT-UniFi-MIB \
+  $(MIBDIR)/UBNT-AirMAX-MIB.txt
+
+mib-dir:
+	@mkdir -p -v $(MIBDIR)
+
+$(MIBDIR)/apc-powernet-mib:
+	@echo ">> Downloading apc-powernet-mib"
+	@curl -s -o $(MIBDIR)/apc-powernet-mib -L $(APC_URL)
+
+$(MIBDIR)/ARISTA-ENTITY-SENSOR-MIB:
+	@echo ">> Downloading ARISTA-ENTITY-SENSOR-MIB"
+	@curl -s -o $(MIBDIR)/ARISTA-ENTITY-SENSOR-MIB -L "$(ARISTA_URL)/ARISTA-ENTITY-SENSOR-MIB.txt"
+
+$(MIBDIR)/ARISTA-SMI-MIB:
+	@echo ">> Downloading ARISTA-SMI-MIB"
+	@curl -s -o $(MIBDIR)/ARISTA-SMI-MIB -L "$(ARISTA_URL)/ARISTA-SMI-MIB.txt"
+
+$(MIBDIR)/ARISTA-SW-IP-FORWARDING-MIB:
+	@echo ">> Downloading ARISTA-SW-IP-FORWARDING-MIB"
+	@curl -s -o $(MIBDIR)/ARISTA-SW-IP-FORWARDING-MIB -L "$(ARISTA_URL)/ARISTA-SW-IP-FORWARDING-MIB.txt"
+
+$(MIBDIR)/.cisco_v2:
+	@echo ">> Downloading cisco v2"
+	@curl -s -L $(CISCO_URL) \
+		| tar -C $(MIBDIR) --strip-components=3 -zxv
+	@touch $(MIBDIR)/.cisco_v2
+
+$(MIBDIR)/mikrotik.mib:
+	@echo ">> Downloading mikrotik"
+	@curl -s -o $(MIBDIR)/mikrotik.mib -L $(MIKROTIK_URL)
+
+$(MIBDIR)/.paloalto_panos:
+	$(eval TMP := $(shell mktemp))
+	@echo ">> Downloading paloalto_pano to $(TMP)"
+	@curl -s -o $(TMP) -L $(PALOALTO_URL)
+	@unzip -j -d $(MIBDIR) $(TMP)
+	@rm -v $(TMP)
+	@touch $(MIBDIR)/.paloalto_panos
+
+$(MIBDIR)/servertech-sentry3-mib:
+	@echo ">> Downloading servertech-sentry3-mib"
+	@curl -s -o $(MIBDIR)/servertech-sentry3-mib -L $(SERVERTECH_URL)
+
+$(MIBDIR)/.synology:
+	$(eval TMP := $(shell mktemp))
```

Comment:
If that fails, `$TMP` is just empty.

Reply:
Make should fail if mktemp returns a non-zero code.